### DISCUSSION TITLE
[Request] permission to bypass on purging playerdata

### DISCUSSION THREAD

**Discussion by momoservertw** [Zrips/CMI#588] [Feature Requests]
**Description of issue or feature request:**
Can you add a permission to bypass on purging playerdata?
For example:
1. cmi.purge.bypass.playerdata - Allows to bypass on purging player data 
2.  cmi.purge.bypass.essentials - Allows to bypass on purging essentials data 

Thank you :)

**Comment by CloudeLecaw** [comment]
why CMI should dedice what essentials datas has to be purged or not? this is the Job of essentials itself not of CMI. Does a permission seems to be the Right way?
not purge when Player X is in a Special Group? so "anti purge permission"? Therefor we Need a hook aggain with the permission System.

For myself.. 
i use a private developed permissionsystem and Solutions like this "not purge by permisison" never will work for me, becaue i think CMI will never support my permissionsystem ;)
(Why Zrips should develop a compatibility to a plugin if just 2 people (me and the dev of it) use it?
 -> Why just two people use it? like i said ist a private development X-D)

Finally to say:
it would more efficent to see a list for antipurgedata where you can enter UUIDs or names (specialy when u just do it on serverstart only), Also a command for adding/removing to/from the list. This would make it simple for owners with differnt ranks:
- setting into a Special group -> execute the cmi command from console to add player to the antipurgedatalist.
- removing group or set to an other group -> execute the cmi command from console that removes the player from the antipurgedatalist.

So you see it is a realy simple to do thing for most owners with a good permissionsystem. ;)

**Comment by momoservertw** [comment]
I would ask author to support it, because CMI already has a purge system, but no bypass way.
In CMI Config, it can purge playerdata, PlayerStats, Essentials...etc.
Maybe adding the features will make this plugin better. Don't you think so?
--
Besides, not everyone has ability to write a plugin.
um... I will try it but unfortunately I busy recently.
I don't have enough time to learn new things. :(

**Comment by CloudeLecaw** [comment]
you are right  about that a antipurge feature would make this feature of cmi better

i just dont like or see why it must be by permission and can not be made by a list?
where players are set on/removed from? CMI cant know all permissionsystems and it can happen that CMI does not know as result if the players have this permission.
May this works for 3-4 permissionsystems but finaly not for all.
and so it is a bit disapointing solution to just use a permission.
What would happen with permission?:
 - The Admins uses the purge command
    -  The CMI hooks into the Permissionfile (so into the permissionsystem) and Looks for all Groups
        with the Bypass - and then forall Players that are effected by this Group.
         - then it starts ist purge with all not listed Players.

.Well that is the blond good working theorie.
-.--and this works as oong CMI can read the permissionsystem. (the file of it)
If Cmi would have a fixed list then you just would need to add/remove a player from the list.
It does not to be an extra file, it also can be a marker or value in the database that allready exist for each player from cmi.  But this all would need the add/delet command for antipruge.
I see no reason why the commands should not be executed when you set your players to VIP or back to fefault...

May i didnt wirte it understandable enough why i think that a permission would be no good idea instead of a list or value..
Well like allwas Zrips can do (and will do what he think what is good, right and better ;) ).So lets wait for the first result :-)

**Comment by LogGits** [comment]
I haven't read this whole thing yet but @CloudeLecaw as long as your plugin supports vaults permission system, CMI can hook into it to read permissions (Granted you are using CMI's vault build).

**Comment by CloudeLecaw** [comment]
i dont use vault… and also not really any Kind of economy, My Server has no reason for using vault.
even if vault is often just called a Bridge, i dont like it, even more then paperspigot xD

**Comment by SmallSansSerif** [comment]
Please add permission to bypass offline purge. Would like to allow purge but have paid ranks be able to avoid getting purged.
